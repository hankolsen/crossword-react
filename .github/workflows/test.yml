name: melodikryzzet-tests

on: 
  push:
    branches:
      - 'master'
  pull_request:


jobs:
  test:
    name: 'Node ${{ matrix.node }}'
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node: [12, 14]
    env:
      SKIP_PREFLIGHT_CHECK: true
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v1
      with: 
        node-version: ${{ matrix.node }}
    - name: Install dependencies
      uses: bahmutov/npm-install@v1
      with: 
        install-command: yarn
    - name: Run tests
      run: yarn test:ci