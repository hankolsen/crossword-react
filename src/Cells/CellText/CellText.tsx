import React from 'react';

import { CELL_HEIGHT, CELL_WIDTH } from 'config';

type Props = {
  column: number;
  row: number;
  text?: string;
};

const CellText = ({ column, row, text = '' }: Props) => (
  <text
    x={(CELL_WIDTH / 2) + (column * (CELL_WIDTH + 1))}
    y={(row * (CELL_HEIGHT + 1)) + 25} // Magic number to align the text vertically...
    className="crossword__cell-text"
    textAnchor="middle"
  >{text}
  </text>
);

export default CellText;
