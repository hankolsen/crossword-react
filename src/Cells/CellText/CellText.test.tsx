import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from 'test/test-utils';
import CellText from 'Cells/CellText/CellText';

describe('CellLabel test', () => {
  it('should not render an empty text', async () => {
    const { container } = render(
      <svg>
        <CellText row={1} column={1} />
      </svg>,
    );
    expect(await screen.queryByText('8')).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render a letter', async () => {
    const { container } = render(
      <svg>
        <CellText row={1} column={1} text="a" />
      </svg>,
    );
    expect(await screen.queryByText(/a/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
