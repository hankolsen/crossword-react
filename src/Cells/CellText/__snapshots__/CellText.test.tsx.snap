// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellLabel test should not render an empty text 1`] = `
<div>
  <div>
    <svg>
      <text
        class="crossword__cell-text"
        text-anchor="middle"
        x="47.5"
        y="57"
      />
    </svg>
  </div>
</div>
`;

exports[`CellLabel test should render a letter 1`] = `
<div>
  <div>
    <svg>
      <text
        class="crossword__cell-text"
        text-anchor="middle"
        x="47.5"
        y="57"
      >
        a
      </text>
    </svg>
  </div>
</div>
`;
